### ISSUE TITLE
Unclear how to detect when file upload is completed

### ISSUE THREAD

**Issue by dbkaplun** [balderdashy/skipper#56]

**Comment by mikermcneil** [comment]
@secrettriangle nothing like that in here yet-- skipper-disk is a good place to start though.  If we come up with a good way of emitting a particular event, we can make that part of the spec for skipper adapters.

**Comment by mikermcneil** [comment]
@secrettriangle  PS- if you have some more details on the use case you're trying to implement, that'll help instruct how we implement the api

**Comment by secrettriangle** [comment]
Well, as I'm using it now, I'm reading directly from `uploadedFile.fd`, which seems to have all the uploaded data stored already, but it's confusing because `status` is `'bufferingOrWriting'` rather than `'completed'` or similar. I'm wondering whether I'm doing it correctly (reading directly within the `.upload` callback) or if I'm overlooking something?

**Comment by juanpasolano** [comment]
I have the same issue I upload a file and I get back the `status: bufferingOrWriting`. And the I try to fetch the file I get a 404 or ENOENT. Which is weird since I would expect the callback to be fired when the file is done uploading and is ready to be served.
I don't know If I am invading @secrettriangle issue but I think my case is quite similar.

**Comment by plato-cambrian** [comment]
+1, my use case is slightly different, I have a form using the default upload adapter to write the upload to `.tmp/background.img`. I am redirecting in the uploadedFiles callback. But I always see the LAST image. If I wait a moment and refresh, I see the new image.

Seems like this callback is fired in between saving to temporary file and moving it to the final location?

**Comment by pjnovas** [comment]
+1 I'm having same exact issue, I expect the callback of upload() to be on completition, but I'm seeing it is not finished when the callback fires.

Is there a way to know when it's done?
Thanks!

**Comment by kamote** [comment]
@pjnovas same issue here.. the only solution i have is to delay the response.to make sure the file is already written

**Comment by sailsbot** [comment]
Thanks for posting, @dbkaplun.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sijo-vijayan** [comment]
I have the same issue same as @pjnovas
